src/Symfony/Component/Translation/Translator.php

```diff
@@ -196,7 +200,26 @@ public function trans($id, array $parameters = array(), $domain = null, $locale
             $domain = 'messages';
         }
 
-        return $this->formatter->format($this->getCatalogue($locale)->get((string) $id, $domain), $locale, $parameters);
+        $id = (string) $id;
+        $catalogue = $this->getCatalogue($locale);
+        $locale = $catalogue->getLocale();
+        $intlDomain = $this->hasIntlFormatter ? $domain.IntlFormatterInterface::DOMAIN_SUFFIX : null;
+        while (true) {
+            if (null !== $intlDomain && $catalogue->defines($id, $intlDomain)) {
+                return $this->formatter->formatIntl($catalogue->get($id, $intlDomain), $locale, $parameters);
+            }
+            if ($catalogue->defines($id, $domain)) {
+                break;
+            }
+            if ($cat = $catalogue->getFallbackCatalogue()) {
```

Comment:
I suppose: this is the exact same code as in transChoice. Not having it here was a bug actually (with no side effects since trans wasn't locale sensitive before)